nodejs/Week3/app.js

```diff
@@ -0,0 +1,48 @@
+import knex from "knex";
+const knexInstance = knex({
+  client: "mysql2",
+  connection: {
+    host: process.env.DB_HOST || "127.0.0.1",
+    port: process.env.DB_PORT || 3306,
+    user: process.env.DB_USER || "root",
+    password: process.env.DB_PASSWORD || "mysql-root-password",
+    database: process.env.DB_NAME || "hyf_node_week3_warmup",
+    multipleStatements: true,
+  },
+});
+
+import express from "express";
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use("/api", apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use("/contacts", contactsAPIRouter);
+
+contactsAPIRouter.get("/", async (req, res) => {
+  let query = knexInstance("contacts");
+  if ("sort" in req.query) {
+    const orderBy = req.query.sort.toString();
+    if (orderBy.length > 0) {
+      query = query.orderBy(orderBy);
+    }
+  }
```

Comment:
hi you need to validate the request here to prevent SQL injection

Reply:
Thank you for your feedback! I've addressed the SQL injection concern by replacing query=query.orderByRaw(orderBy) with query=query.orderBy(orderBy)